### PR TITLE
Support :if for tags

### PR DESCRIPTION
**Pull request by chrismccord**
Still needs docs but ready for review

### CODE REVIEW

**Comment by chrismccord** [comment]
I also think we should support this for slots if we can, but I know there are some considerations. The mechanism by which it would work would have to be different internally than this PR, but we should be able to easily filter the slot collection before calling the function component.

File: lib/phoenix_live_view/html_engine.ex [hunk 1]
```diff
@@ -600,32 +608,38 @@ defmodule Phoenix.LiveView.HTMLEngine do
...
-      {expr, outer_state} ->
-        handle_for_expr(state, expr, outer_state)
-
-      nil ->
-        state
+    if tag_open_meta[:for] || tag_open_meta[:if] do
+      handle_special_expr(state, tag_open_meta)
+    else
+      state
     end
```
Comment (josevalim) [inline lib/phoenix_live_view/html_engine.ex:643]: Might as well remove this conditional here and add a catch-all clause to `handle_special_expr`?

```suggestion
    handle_special_expr(state, tag_open_meta)
```

**Review by josevalim — APPROVED** [review]
Beautiful. Just one minor comment, docs, and we can ship it :)

**Comment by connorlay** [comment]
Once this is merged I'll open a PR for tree-sitter-heex adding support for :if 🙂

**Comment by connorlay** [comment]
PR for tree-sitter-heex is ready for review: https://github.com/phoenixframework/tree-sitter-heex/issues/19

**Pull request merged**